src/transformers/models/deformable_detr/modeling_deformable_detr.py

```diff
@@ -1582,7 +1582,7 @@ def get_proposal_pos_embed(self, proposals):
         temperature = 10000
         scale = 2 * math.pi
 
-        dim_t = torch.arange(num_pos_feats, dtype=torch.int64, device=proposals.device).float()
+        dim_t = torch.arange(num_pos_feats, dtype=torch.int64, device=proposals.device).to(proposals.dtype)
```

Comment:
Can't we pass in the dtype directly here? 

```suggestion
        dim_t = torch.arange(num_pos_feats, dtype=proposals.dtype, device=proposals.device)
```

Reply:
Possible. Thank you